relay-config/src/config.rs

```diff
@@ -1002,6 +1013,20 @@ pub enum EnvelopeSpoolVersion {
     V2,
 }
 
+/// The strategy to use for envelope buffering.
+#[derive(Debug, Default, Deserialize, Serialize)]
+#[serde(rename_all = "snake_case")]
+pub enum EnvelopeBufferStrategy {
+    /// Use an in-memory buffer for envelopes.
+    #[default]
+    Memory,
```

Comment:
With this default, our currently deployed configs will revert to `Memory` even if they have a `path` set.

Bonus points if you can make it an adjacently or untagged enum while maintaining backward compatibility:

```rust
Memory,
Sqlite(PathBuf),
FileBacked(PathBuf)
```

Reply:
Yes, this is a good point. I will try to make it work.